File: 卡码/代码随想录/JS/二叉树/01-二叉树遍历方式/LeetCode_0107.js
//107. 二叉树的层序遍历 II
import TreeNode from "../TreeNode.js";

const levelOrderBottom = function(root) {
  const res=new Array();
  const queue=new Array();
  let size=0;
  if(root) queue.push(root);
  while(queue.length>0){
    size=queue.length;
    const v=new Array();
    while(size-->0){
      const node=queue.shift();
      v.push(node.val);
      if(node.left) queue.push(node.left);
      if(node.right) queue.push(node.right);
    }
    res.unshift(v);
  }
  return res;
};

File: 卡码/代码随想录/Go/二叉树/01-二叉树遍历方式/LeetCode_0107.go
//107. 二叉树的层序遍历 II
package main

import (
	"slices"
	. "binaryTree/types"
)

func levelOrderBottom(root *TreeNode) [][]int {
	queue:=make([]*TreeNode,0)
	res:=make([][]int,0)
	if root!=nil {
		queue=append(queue,root)
	}
	var size int
	for len(queue)>0 {
		size=len(queue)
		v:=make([]int,0)
		for size>0 {
			size--
			node:=queue[0]
			queue=queue[1:]
			v=append(v,node.Val)
			if node.Left!=nil {
				queue=append(queue,node.Left)
			}
			if node.Right!=nil {
				queue=append(queue,node.Right)
			}
		}
		res=append(res,v)
	}
	slices.Reverse(res)
	return res
}
